src/ansi-c/library/string.c

```diff
@@ -554,7 +554,20 @@ __CPROVER_HIDE:;
 
 #undef strlen
 
-inline size_t strlen(const char *s)
+inline size_t strlen(const char *s)  
+  __CPROVER_requires(s != NULL && 
+    __CPROVER_exists {
+      int i;
+      (0 <= i && i < MAX_BUFFER_SIZE) ==> (i < __CPROVER_OBJECT_SIZE(s)) ==> s[i] == '\0'
```

Comment:
`MAX_STRING_SIZE`would be more appropriate.

Reply:
I think neither of those actually belong within CBMC; both come from our harnesses. We have two options and I am not sure which is better:

1. Get rid of these constant bounds and write quantified expressions for the SMT backend, so simply:
```c
      (0 <= i && i < __CPROVER_OBJECT_SIZE(s)) ==> s[i] == '\0'
```
2. Define some constant bounds like `MAX_STRING_SIZE` for CBMC library functions, but then those would have to be set in addition to, and be consistent with, the harness-specific constants. This also raises more questions: How many such constant bounds to introduce into CBMC, like `MAX_BUFFER_SIZE` for `malloc` as well? What if the user wants to use two different bounds for different allocations? We have for example two bounds for some of the array functions: a MAX_ARRAY_SIZE and a MAX_ARRAY_ELEMENT_SIZE.